src/Compilers/VisualBasic/Portable/Symbols/Source/CustomModifierUtils.vb

```diff
@@ -2,9 +2,65 @@
 
 Imports System.Collections.Immutable
 Imports System.Runtime.InteropServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
 
 Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
     Friend Class CustomModifierUtils
+
+        ''' <summary>
+        ''' sourceMethod has the custom modifiers
+        ''' </summary>
+        Friend Shared Sub CopyMethodCustomModifiers(
+                                sourceMethod As MethodSymbol,
+                                destinationTypeParameters As ImmutableArray(Of TypeSymbol),
+                                <[In], Out> ByRef destinationReturnType As TypeSymbol,
+                                containingAssembly As AssemblySymbol,
+                                <[In], Out> ByRef parameters As ImmutableArray(Of ParameterSymbol))
+
+            Debug.Assert(sourceMethod IsNot Nothing)
+
+            ' For the most part, we will copy custom modifiers by copying types.
+            ' The only time when this fails Is when the type refers to a type parameter
+            ' owned by the overridden method.  We need to replace all such references
+            ' with (equivalent) type parameters owned by this method.  We know that
+            ' we can perform this mapping positionally, because the method signatures
+            ' have already been compared.
+            Dim constructedMethod As MethodSymbol = sourceMethod.ConstructIfGeneric(destinationTypeParameters)
+
+            parameters = CustomModifierUtils.CopyParameterCustomModifiers(constructedMethod.Parameters, parameters)
+
+            Dim destinationReturnTypeWithCustomModifiers = constructedMethod.ReturnType
```

Comment:
>destinationReturnTypeWithCustomModifiers [](start = 16, length = 40)

This is actually a type from source method, but he name says destination. This is very confusing.